mrp_subcontracting_inhibit/models/purchase_order.py

```diff
@@ -0,0 +1,9 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
```

Comment:
I'm thinking that this should be at PO line level, as you may want some products with subcontracting and some of them without.